crates/sui-graphql-client/src/lib.rs

```diff
@@ -0,0 +1,568 @@
+    page_info: PageInfo,
+    /// The data returned by the server.
+    nodes: Vec<T>,
+}
+
+impl<T> Page<T> {
+    pub fn page_info(&self) -> &PageInfo {
+        &self.page_info
+    }
+
+    pub fn nodes(&self) -> &Vec<T> {
+        &self.nodes
+    }
+
+    fn new(page_info: PageInfo, nodes: Vec<T>) -> Self {
+        Self { page_info, nodes }
+    }
+}
+
+/// The SuiClient is a GraphQL client for interacting with the Sui blockchain.
+/// By default, it uses Reqwest as the HTTP client.
+pub struct SuiClient {
+    /// The URL of the GraphQL server.
+    rpc: String,
+    /// GraphQL supports multiple versions of the service. By default, the client uses the stable
+    /// version of the service.
+    rpc_version: Option<String>,
+    inner: ReqwestHttpClient,
+}
+
+impl Default for SuiClient {
+    /// Create a new [`SuiClient`] with testnet as the default GraphQL server.
+    fn default() -> Self {
+        Self {
+            rpc: TESTNET_HOST.to_string(),
+            rpc_version: None,
+            inner: ReqwestHttpClient::new(),
+        }
+    }
+}
+
+impl SuiClient {
+    // ===========================================================================
+    // Client Misc API
+    // ===========================================================================
+
+    /// Set the server address for the GraphQL GraphQL client. It should be a valid URL with a host and
+    /// optionally a port number.
+    pub fn set_rpc_server(&mut self, server: &str) -> Result<(), anyhow::Error> {
+        reqwest::Url::parse(server)?;
+        self.rpc = server.to_string();
+        Ok(())
+    }
+
+    /// Set the version for the GraphQL GraphQL client. The default set version is stable.
+    ///
+    /// Use the `service_config` API to get the `available_versions` field to know which versions
+    /// are supported by this service.
+    pub fn set_version(&mut self, version: Option<&str>) {
+        self.rpc_version = version.map(|v| v.to_string());
+    }
+
+    /// Return the URL for the GraphQL GraphQL client after checking if the version is set.
+    fn url(&self) -> String {
+        if let Some(version) = &self.rpc_version {
+            format!("{}/{}", self.rpc, version)
+        } else {
+            self.rpc.to_string()
+        }
+    }
+
+    /// Set the GraphQL to localhost and port 9125.
+    pub fn set_localhost(&mut self) -> &Self {
+        self.rpc = DEFAULT_LOCAL_HOST.to_string();
+        self
+    }
+
+    /// Set GraphQL client to devnet.
+    pub fn set_devnet(&mut self) -> &Self {
+        self.rpc = DEVNET_HOST.to_string();
+        self.set_version(None);
+        self
+    }
+
+    /// Set GraphQL client to testnet.
+    pub fn set_testnet(&mut self) -> &Self {
+        self.rpc = TESTNET_HOST.to_string();
+        self
+    }
+
+    /// Set GraphQL client to mainnet.
+    pub fn set_mainnet(&mut self) -> &Self {
+        self.rpc = MAINNET_HOST.to_string();
+        self
+    }
```

Comment:
Generally methods that are of the form `set_*` won't return anything. It sort of seems like you're half-way between wanting a Builder pattern, in which case you'd want to return either `Self` or `&mut Self` so that further changes could be made via chaining.